java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture3/tasks/_2/Number2.java

```diff
@@ -0,0 +1,193 @@
+package com.zoolatech.lecture3.tasks._2;
+
+/*
+Design a class that represents a user account validator.
+The validator needs to use a list of smaller validators to perform a validation of specific account fields.
+All validators need to have a method isValid that accepts a user account object and returns either true or false.
+The smaller validators need to accept a field name during the creation
+and apply a specific validation rule to this field.
+A user account validator needs to return true if all smaller validators returned true,
+and false otherwise. List of user account fields:
+userId (number), firstName(string), lastName(string),
+country(string), userEmail (string), phoneNumber(string),
+birthDay (number), birthMonth (number).
+List of validation rules:
+string is not empty, valid email, valid phone number, number is in the range between X and Y,
+number is bigger than a value X, number is less than a value X.
+You don’t need to perform actual validation, it’s enough to print smth like
+“validating that fieldName is less than 10” and return true from a smaller validator.
+In the main method, create a user account and user account validator objects.
+The validator needs to contain all validation rules that can be applied to user account objects.
+*/
+
+import java.util.Scanner;
+
+public class Number2 {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        Account account = new Account();
+        Validator validator = new Validator();
+        System.out.println("Type user id:");
+        account.userId = scanner.nextInt();
+        System.out.println("Type user name:");
+        account.firstName = scanner.next();
+        System.out.println("Type user last name:");
+        account.lastName = scanner.next();
+        System.out.println("Type your country");
+        System.out.println("Ukraine, USA, Canada, Mexico, Poland, Germany, France, Italy, Turkey");
+        account.country = scanner.next();
+        System.out.println("Type user email:");
+        account.userEmail = scanner.next();
+        System.out.println("Type user phone number:");
+        account.phoneNumber = scanner.next();
+        System.out.println("Type birth day");
+        account.birthDay = scanner.nextInt();
+        System.out.println("And birthday month");
+        account.birthMonth = scanner.nextInt();
+        validator.isValid(account);
+    }
+}
+
+interface Verification {
+    public boolean isValid(Account account);
+}
+
+class Account {
```

Comment:
don't forget regarding the encapsulation - instance fields should be accessed through methods

you can also convert this class to a record